pkg/reader/ccip.go

```diff
@@ -868,32 +865,26 @@ func (r *ccipChainReader) GetRmnCurseInfo(
 	lggr.Debugw("got cursed subjects", "cursedSubjects", cursedSubjects.CursedSubjects)
 
 	return getCurseInfoFromCursedSubjects(
-		lggr,
 		mapset.NewSet(cursedSubjects.CursedSubjects...),
 		r.destChain,
-		sourceChainSelectors,
 	), nil
 }
 
 func getCurseInfoFromCursedSubjects(
-	lggr logger.Logger,
 	cursedSubjectsSet mapset.Set[[16]byte],
 	destChainSelector cciptypes.ChainSelector,
-	sourceChainSelectors []cciptypes.ChainSelector,
 ) *CurseInfo {
 	curseInfo := &CurseInfo{
-		CursedSourceChains: make(map[cciptypes.ChainSelector]bool, len(sourceChainSelectors)),
+		CursedSourceChains: make(map[cciptypes.ChainSelector]bool, cursedSubjectsSet.Cardinality()),
 		CursedDestination: cursedSubjectsSet.Contains(GlobalCurseSubject) ||
 			cursedSubjectsSet.Contains(chainSelectorToBytes16(destChainSelector)),
 		GlobalCurse: cursedSubjectsSet.Contains(GlobalCurseSubject),
 	}
 
-	for _, ch := range sourceChainSelectors {
-		chainSelB16 := chainSelectorToBytes16(ch)
-		lggr.Debugf("checking if chain %d is cursed after casting it to 16 bytes: %v", ch, chainSelB16)
-		curseInfo.CursedSourceChains[ch] = cursedSubjectsSet.Contains(chainSelB16)
+	for _, cursedSource := range cursedSubjectsSet.ToSlice() {
```

Comment:
This is the only change. Instead of converting chainSelector to uint64, I'm doing the opposite - read RMN response and cast to chain selectors

Reply:
Interesting that none of the tests catch that :P

Reply:
Sure, renamed